### PR TITLE
Feature/ui/image viewer fix

### PR DESCRIPTION
**Pull request by sir2805**

### CODE REVIEW

File: slide_analysis/UI/view/image_viewer.py [hunk 1]
```diff
@@ -20,58 +22,39 @@ def __init__(self, controller, model):
         super(ImageViewer, self).__init__()
         self.model = model
         self.controller = controller
-
         self.image_helper = None
```
Comment (sir2805) [inline slide_analysis/UI/view/image_viewer.py]: move to fullslide_viewer

File: slide_analysis/UI/view/fullslide_viewer.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import time
+from PyQt5.QtCore import Qt, QRectF
+from PyQt5.QtGui import QPixmap
+from PyQt5.QtWidgets import QGraphicsView, QGraphicsScene, QFrame, QGraphicsPixmapItem
+
+from slide_analysis.UI.view.tile_view_widget import TilePreviewPopup
+
+
+class FullslideViewer(QGraphicsView):
```
Comment (sir2805) [inline slide_analysis/UI/view/fullslide_viewer.py]: rename to ImageDisplay

File: slide_analysis/UI/view/fullslide_viewer.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                    print('move to next image level ' + str(millis2 - millis1))
+
+                    pixmap = QPixmap.fromImage(self.image_helper.get_current_image())
+                    millis1 = round(time.time() * 1000)
+                    self._photo.setPixmap(pixmap)
+                    millis2 = round(time.time() * 1000)
+                    print('set new pixmap ' + str(millis2 - millis1))
+                    millis1 = round(time.time() * 1000)
+                    self.update_image()
+                    millis2 = round(time.time() * 1000)
```
Comment (sir2805) [inline slide_analysis/UI/view/fullslide_viewer.py]: remove profiling

File: slide_analysis/UI/view/image_helper.py [hunk 4]
```diff
@@ -1,34 +1,40 @@
...
-        print(self.image_dimensions)
         self.print_status()
 
-    def get_tile_coordinates(self, mouse_pos_point, scroll_area_coordinates):
-        # scroll_area_coordinates = self.scrollArea.geometry()
-        actual_coordianates_x = (mouse_pos_point.x() - scroll_area_coordinates.x()) * self.current_window_size[
-            0] // scroll_area_coordinates.width() * pow(2, self.current_level) + self.current_coordinates[0]
-        actual_coordianates_y = (mouse_pos_point.y() - scroll_area_coordinates.y()) * self.current_window_size[
-            1] // scroll_area_coordinates.height() * pow(2, self.current_level) + self.current_coordinates[1]
+    def get_filepath(self):
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py:31]: rename

File: slide_analysis/UI/view/image_helper.py [hunk 5]
```diff
@@ -1,34 +1,40 @@
...
 
 class ImageHelper:
-    def __init__(self, filepath):
-        self.filepath = filepath
-        self.openslide_image = openslide.OpenSlide(filepath)
-        self.current_level = self.openslide_image.level_count - 1
+    def __init__(self, filename):
+        self.filename = filename
+        self.openslide_image = openslide.OpenSlide(filename)
+        self.max_level = self.openslide_image.level_count - 1
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py]: remove

File: slide_analysis/UI/view/image_helper.py [hunk 6]
```diff
@@ -1,34 +1,40 @@
...
+        self.current_level = self.max_level
         self.level_dimensions = self.openslide_image.level_dimensions
-        self.current_coordinates = (0, 0)
+        self.current_displayed_image_part_coordinates = (0, 0)
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py]: rename to current_cached_image_coordinates

File: slide_analysis/UI/view/image_helper.py [hunk 7]
```diff
@@ -1,34 +1,40 @@
...
+        self.current_image_part_coordinates = self.current_displayed_image_part_coordinates
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py]: rename

File: slide_analysis/UI/view/image_helper.py [hunk 8]
```diff
@@ -1,34 +1,40 @@
...
+        self.current_displayed_image_part_size = self.level_dimensions[self.current_level]
+        self.current_image_part_size = self.current_displayed_image_part_size
+
         self.image_dimensions = self.level_dimensions[0]
         self.scale_factor = BASE_SCALE_FACTOR
         self.current_movement_step = (self.image_dimensions[0] // self.scale_factor,
                                       self.image_dimensions[1] // self.scale_factor)
-        self.current_window_size = self.level_dimensions[self.current_level]
-        self.image = self.openslide_image.read_region(self.current_coordinates, self.current_level,
+        self.image = self.openslide_image.read_region(self.current_displayed_image_part_coordinates, self.current_level,
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py]: change current_displayed_image_part_coordinates to current_image_part_coordinates

File: slide_analysis/UI/view/image_helper.py [hunk 9]
```diff
@@ -45,54 +51,164 @@ def __calculate_movement_step_coordinates(self):
...
-                             self.current_coordinates[1])
-
-    def move_down(self):
-        self.set_coordinates(self.current_coordinates[0],
-                             self.current_coordinates[1] + self.current_movement_step[1])
-
-    def move_up(self):
-        self.set_coordinates(self.current_coordinates[0],
-                             self.current_coordinates[1] - self.current_movement_step[1])
+        return self.update_q_image()
```
Comment (sir2805) [inline slide_analysis/UI/view/image_helper.py]: remove

**Review by Vozf — APPROVED** [review]

**Pull request merged**